src/components/Personalization/decisionsExtractor.js

```diff
@@ -75,11 +77,35 @@ const extractDecisionsByScope = (decisions, scope) => {
   return { pageWideScopeDecisions, nonPageWideScopeDecisions };
 };
 
+const getGroupedDecisions = unprocessedDecisions => {
+  const decisionsGroupedByRedirectItemSchema = splitDecisions(
+    unprocessedDecisions,
+    REDIRECT_ITEM
+  );
+  const decisionsGroupedByDomActionSchema = splitDecisions(
+    decisionsGroupedByRedirectItemSchema.unmatchedDecisions,
+    DOM_ACTION
+  );
+
+  const {
+    pageWideScopeDecisions,
+    nonPageWideScopeDecisions
+  } = extractDecisionsByScope(
+    decisionsGroupedByDomActionSchema.matchedDecisions,
+    PAGE_WIDE_SCOPE
+  );
+
+  return {
+    redirectDecisions: decisionsGroupedByRedirectItemSchema.matchedDecisions,
+    pageWideScopeDecisions,
+    viewDecisions: nonPageWideScopeDecisions,
+    formBasedComposedDecisions:
+      decisionsGroupedByDomActionSchema.unmatchedDecisions
+  };
+};
+
 export default {
-  groupDecisionsBySchema({ decisions, schema }) {
-    return splitDecisions(decisions, schema);
-  },
-  groupDecisionsByScope({ decisions, scope }) {
-    return extractDecisionsByScope(decisions, scope);
+  groupDecisions(decisions) {
+    return getGroupedDecisions(decisions);
```

Comment:
I really like this change. I think it cleaned up a lot of the code that depended on it. Consider exporting this function as the default and renaming the module accordingly.

Reply:
thank you :)